app/controllers/api/v1/offer_responses_controller.rb

```diff
@@ -19,27 +19,22 @@ class  OfferResponsesController < Api::V1::ApiController
         end
       end
 
-      api :POST, "/v1/offer_responses","Create a offer Response"
+      api :POST, "/v1/offer_responses", "Create a Offer Response"
       def create
-        user_id, offer_id = params["offer_response"].values_at('user_id', 'offer_id')
-        return  render json: { errors: "Offer and User Id required"}, status: 422 unless user_id.present? && offer_id.present?
+        record = OfferResponse.find_or_create_by(offer_responses_params)
```

Comment:
`CanCanCan` already gives you `@offer_response` with the params loaded, you don't need to do a `find_or_create_by` here. Just do `if @offer_response.save`

Reply:
+ 1